homeassistant/helpers/service.py

```diff
@@ -86,33 +87,67 @@ def call_from_config(hass, config, blocking=False, variables=None,
 
 
 @bind_hass
-def extract_entity_ids(hass, service_call, expand_group=True):
+async def extract_entity_ids(hass, service_call, expand_group=True):
     """Extract a list of entity ids from a service call.
 
     Will convert group entity ids to the entity ids it represents.
-
-    Async friendly.
     """
-    if not (service_call.data and ATTR_ENTITY_ID in service_call.data):
-        return []
+    return run_coroutine_threadsafe(
```

Comment:
We're within the event loop/coroutine, doesn't it raise a RuntimeError?

Reply:
That should not have been async.  Copy paste error.

Reply:
We might even be able to remove this method completely, but I don't know if any custom component is using it.